consensus/enclave/impl/src/lib.rs

```diff
@@ -462,6 +462,7 @@ impl ConsensusEnclave for SgxConsensusEnclave {
             BLOCK_VERSION,
             &parent_block.id,
             parent_block.index + 1,
+            parent_block.cumulative_txo_count,
```

Comment:
Maybe we should instead just pass the parent block here? I find it confusing that we are passing the parent's id, our index, and then the parent's cumulative tx count. I know it means redoing a bunch of changes but it would be much cleaner following that.

Reply:
yah i thought about this too

So one question I have is, what if I want to partially download the blockchain?

If the only way to make a new block is to either already have a block, or to have an origin block, then that seems like I will have to download the whole blockchain

So maybe there should be like a "BlockContext" or "BlockParent" or something that captures only what you need to know about the parent to make the next block?

Reply:
If you're making new blocks then you are a consensus service node need all of the blockchain. I don't think we need to support making new blocks if you don't have the blockchain.

Reply:
Also, the `Block` doesn't actually have its contents - only ID, hash, parent ID, etc.

Reply:
So like, suppose I want to download only blocks 200 - 500 of the block chain. How does it work? I just deserialize them, and then I have them?

Reply:
Hm, I guess we would still need a constructor that takes the individual values for the proto->native rust conversion, you're right. But it looks like most places where we form a block do have the parent available.

Reply:
Might be OK to have an additional constructor, like Block::new_from_parent, which is just syntactic sugar over the more verbose Block::new

Reply:
Yeah, thats what I'm thinking.